nox/sessions.py

```diff
@@ -177,6 +178,19 @@ def __init__(self, name, signature, func, global_config):
 
     def _create_config(self):
         self.config = SessionConfig(posargs=self.global_config.posargs)
+
+        # By default, nox should quietly change to the directory where
+        # the nox.py file is located; however, do not be overly stringent
+        # about this. If we cannot easily inspect that location, just give up
+        # and skip that command.
+        try:
+            cwd = os.path.dirname(inspect.getsourcefile(self.func))
```

Comment:
The noxfile itself is available in the `global_config` object: https://github.com/jonparrott/nox/blob/3f6d1df8d613c76b0cc7a5aec4797d389b3b86da/nox/main.py#L142 any reason not to use that directly?